client/my-sites/comments/comment/comment-link.jsx

```diff
@@ -3,55 +3,32 @@
  */
 import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
-import { connect } from 'react-redux';
-import { get, includes, omit } from 'lodash';
+import page from 'page';
 
-/**
- * Internal dependencies
- */
-import { navigate } from 'calypso/state/ui/actions';
-
-export class CommentLink extends PureComponent {
+export default class CommentLink extends PureComponent {
 	static propTypes = {
-		children: PropTypes.any,
 		commentId: PropTypes.number,
 		href: PropTypes.string,
-		navigate: PropTypes.func,
 	};
 
 	handleClick = ( event ) => {
-		if ( ! window ) {
-			return;
-		}
 		event.preventDefault();
 		window.scrollTo( 0, 0 );
 
 		const { commentId, href } = this.props;
-		const path = get( window, 'history.state.path' );
+		const { path } = window.history.state;
 
-		const newPath = includes( path, '#' )
+		const newPath = path.includes( '#' )
 			? path.replace( /[#].*/, `#comment-${ commentId }` )
 			: `${ path }#comment-${ commentId }`;
 
 		window.history.replaceState( { ...window.history.state, path: newPath }, null );
 
-		this.props.navigate( href );
+		page( href );
```

Comment:
If we are doing both a `page()` and a `window.history.replaceState()`, should we just use `navigate()`?

Reply:
Honestly I have no idea how this is supposed to work 🙂 Even after playing with the UI. When navigating to a comment, there are two options where to navigate:
- if the comment is also on the current page (typical for a post's comment list), just scroll to the `#comment-123` anchor on the current page, which is something like `/comments/all/:site/:post_id`
- or navigate to a dedicated route just for this comment: `/comments/:site/:comment_id`

This click handler tries to do both at once 😕: set the anchor with `replaceState`, and then navigate to comment-specific route with `page( href )`. Maybe @Copons remembers how this was supposed to work in 2018?

My uncertainty about these comments are the main reason why I still keep this as Draft.

Reply:
I see. I guess in any case it makes sense to keep this as close to how it was before, unless we find a regression here (which I don't) 👍